Document use of SPIRV-Tools with asm output (#1342)
KhronosGroup/SPIRV-LLVM-Translator: README.md

  moment, see [KhronosGroup/SPIRV-LLVM-Translator#477](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/issues/477)
  to track progress, discuss and contribute.

### Build with SPIRV-Tools

> The translator can use SPIRV-Tools to generate assembly with conformant syntax.

Reviewer: It would be nice to have a link to the SPIR-V Tools repo.
```suggestion
The translator can use [SPIRV-Tools](https://github.com/KhronosGroup/SPIRV-Tools) to generate assembly with conformant syntax.
```

> The translator can use [SPIRV-Tools](https://github.com/KhronosGroup/SPIRV-Tools) to generate assembly with conformant syntax.

Author: ```suggestion
The translator can use [SPIRV-Tools](https://github.com/KhronosGroup/SPIRV-Tools) to generate assembly with widely adopted syntax.
```

If SPIRV-Tools have been installed prior to the build it will be detected and
used automatically. However it is also possible to enable use of SPIRV-Tools
from a custom location using the following instructions:

1. Checkout, build and install SPIRV-Tools using
   [the following instructions](https://github.com/KhronosGroup/SPIRV-Tools#build).

>    Example using cmake:

Author: ```suggestion
   Example using CMake with Ninja:
```

```

> cmake -G Ninja .. -DCMAKE_INSTALL_PREFIX=<location to installation>

Author: ```suggestion
cmake [-G <platform-generator>] <SPIRV-Tools location> -DCMAKE_INSTALL_PREFIX=<installation location>
```

ninja install
```

> 2. Using pkg-config when building the translator by setting

Author: ```suggestion
2. Use pkg-config when building the translator by setting
```

Author: I am not sure what part is not correct - it is saying that by means of pkg-config we can specify the SPIRV-Tools installation...

Reviewer 2: Maybe rephrase it as:

```suggestion
2. Point pkg-config to the SPIR-V tools installation when configuring the translator by setting
```

>    `PKG_CONFIG_PATH=<location to installation>/lib/pkgconfig/` variable

Author: ```suggestion
   `PKG_CONFIG_PATH=<SPIRV-Tools installation location>/lib/pkgconfig/` variable
```

   before the cmake line invocation.
   Example:
```

> PKG_CONFIG_PATH=<location to installation>/lib/pkgconfig/ cmake ...

Author: ```suggestion
PKG_CONFIG_PATH=<location to installation>/lib/pkgconfig/ cmake <other options>
```

```

> To test the correctness of translator build run the following line

Reviewer 2: ```suggestion
To verify the SPIR-V Tools integration in the translator build, run the following line
```

    * `-o file_name` - to specify output name
    * `-spirv-debug` - output debugging information
    * `-spirv-text` - read/write SPIR-V in an internal textual format for debugging purpose. The textual format is not defined by SPIR-V spec.

>     * `--spirv-tools-dis` - print SPIR-V assembly in standard format. Only available on [builds with SPIR-V](#build-with-spirv-tools).

Reviewer: I don't think there is a really _standard_ format for textual representation of SPIR-V.

Also a typo in the link
```suggestion
    * `--spirv-tools-dis` - print SPIR-V assembly in standard format. Only available on [builds with SPIRV-Tools](#build-with-spirv-tools).
```

Author: Hmm, although I thought that SPIRV-Tools produce the standard syntax. I am using this terminology in clang docs too. I think it will be more confusing if we start talking about SPIRV-Tools syntax because then it is not clear why would anyone bother about some syntax from another project. We should bear in mind that this documentation is not for top experts in the tooling aspects of this area.

Author: ok, I had the impression that SPIRV-Tools produced a defacto standard syntax because it is used in the examples in the spec... but if we don't align on this then I would say we should not write overstatements in the public docs.

Let's go for` SPIRV-Tools syntax` for now even though I don't like exposing the implementation details to the users... but at least it is not an incorrect statement.
